src/test/java/uk/gov/digital/ho/hocs/info/api/TeamIntegrationTests.java

```diff
@@ -30,133 +26,109 @@
 import uk.gov.digital.ho.hocs.info.security.AccessLevel;
 import uk.gov.digital.ho.hocs.info.security.Base64UUID;
 import uk.gov.digital.ho.hocs.info.security.KeycloakService;
+import uk.gov.digital.ho.hocs.info.utils.BaseKeycloakTest;
 
-import java.io.File;
 import java.io.IOException;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.UUID;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.springframework.http.HttpMethod.GET;
-import static org.springframework.test.annotation.DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD;
 import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
 import static org.springframework.test.context.jdbc.SqlConfig.TransactionMode.ISOLATED;
 import static org.springframework.test.web.client.MockRestServiceServer.bindTo;
 import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
 import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
 import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
 
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 @Sql(scripts = "classpath:beforeTest.sql", config = @SqlConfig(transactionMode = ISOLATED))
 @Sql(scripts = "classpath:afterTest.sql", config = @SqlConfig(transactionMode = ISOLATED), executionPhase = AFTER_TEST_METHOD)
-@DirtiesContext(classMode = AFTER_EACH_TEST_METHOD)
 @ActiveProfiles({"local", "integration"})
-public class TeamIntegrationTests {
-
-    TestRestTemplate testRestTemplate = new TestRestTemplate();
+public class TeamIntegrationTests extends BaseKeycloakTest {
 
     @Autowired
     private RestTemplate restTemplate;
 
     @Autowired
-    TeamRepository teamRepository;
+    private TeamRepository teamRepository;
 
     @Autowired
-    KeycloakService keycloakService;
+    private ObjectMapper mapper;
 
-    Keycloak keycloakClient;
+    @LocalServerPort
+    private int port;
 
+    private TestRestTemplate testRestTemplate = new TestRestTemplate();
     private MockRestServiceServer mockCaseworkService;
-
     private HttpHeaders headers;
-
-    @LocalServerPort
-    int port;
-
-    @Autowired
-    ObjectMapper mapper;
-
-    @Value("${keycloak.server.url}")
-    String serverUrl;
-    @Value("${keycloak.username}")
-    String username;
-    @Value("${keycloak.password}")
-    String password;
-    @Value("${keycloak.client.id}")
-    String clientId;
-    @Value("${keycloak.realm}")
-    String HOCS_REALM;
-
     private String userId;
     private final UUID unitUUID = UUID.fromString("09221c48-b916-47df-9aa0-a0194f86f6dd");
 
+    @BeforeEach
```

Comment:
We need to clear the realm down for any tests that edit the realm.  I'd rather just dump the realm then work out what needs removing. 

It's relatively quick and when we have the user/team service it'll be negligible anyway.

Reply:
We have dirties context on, so it recreates the bean, unsure if storing it in the a BeforeAll would cause a weird memory pointer problem. 

I can have a look though.